### PR TITLE
performance improvements, use onpush

### PR DESCRIPTION
**Pull request by jonnhy12333**
turns out that most of our client side performance problems was due to the angular checks that were being done constantly.  I just turned it to 'OnPush' instead of using the default change detection strategy and it helped a ton (nothing is broken that I'm aware of).

I included some code in main.ts so that you can easily turn on debugging (in Dev only) in the future to check that your changes don't increase the change detection time (should always be below 3ms)

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
 
 import 'hammerjs';
+import { enableDebugTools } from '@angular/platform-browser';
 
 if (environment.production) {
   enableProdMode();
 }
 
 platformBrowserDynamic().bootstrapModule(AppModule)
+  .then(moduleRef => {
```
Comment (greggiannone) [inline src/main.ts]: Should this be commented out like it says in the PR comment?

File: src/app/components/user-list/user-list.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="user-list" *ngIf="users$ | async | userOrderBy : 'status' as users">
+<div class="user-list" *ngIf="users$ | async | userOrderBy as users">
```
Comment (greggiannone) [inline src/app/components/user-list/user-list.component.html]: What's dis for?

File: src/main.ts (hunk 1 above)
Comment (jonnhy12333) [reply src/main.ts]: Oh whoops, I should update that comment, figured I don't actually need to comment it out if it's only in dev

Comment (jonnhy12333) [reply src/main.ts]: Oh lol you meant PR comment, I should update that

**Pull request merged**